File: Player.cpp
#include "Player.h"

Player:: Player() {

}

Player:: Player(const char &c) {
	this->c_ = c;
}
char Player:: getPlayerChar() {
	return this->c_;
}
/**
 * calls the move function of the game parameter.
 */
void Player:: makeMove(int i , int j , GamePlay * game) {
	game->move(this->c_ , i , j);
}

Player& Player:: operator= (const Player& p) {
	return *this;
}

Player:: Player(const Player &p) {
	this->c_ = p.c_;
}

Player:: ~Player() {
}

File: java for ex6/Player.java
public abstract class Player {
	protected char c;

	public Player(char ch) {
		this.c = ch;
	}

	public char getChar() {
		return this.c;
	}

	public abstract boolean playOneTurn(GamePlay game);

	public void makeMove(int i , int j , GamePlay game) {
		game.move(this , i , j);
	}
}